data-prepper-expression/src/main/java/org/opensearch/dataprepper/expression/LengthExpressionFunction.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.expression;
+
+import java.util.List;
+import javax.inject.Named;
+
+@Named
+public class LengthExpressionFunction implements ExpressionFunction {
+    public String getFunctionName() {
+        return "length";
+    }
+
+    public Object evaluate(final List<Object> args) {
+        if (args.size() != 1) {
+            throw new RuntimeException("length() takes only one argument");
+        }
+        Object arg = args.get(0);
+        if (!(arg instanceof String)) {
+            throw new RuntimeException("length() takes only String type arguments");
+        }
+        String argStr = ((String)arg).trim();
+        if (argStr.length() == 0) {
+            return 0;
+        }
+        if (argStr.charAt(0) == '\"') {
+            if (argStr.charAt(argStr.length()-1) != '\"') {
+                throw new RuntimeException("Invalid string passed to length() "+argStr);
+            }
+            return Integer.valueOf(argStr.length()-2);
```

Comment:
Does excluding the extra quotes make sense to the user excluding only the first set of extra quotes? How would this work for string data like: `"\"\"\"\"foobar\"\"\"\""`?

Reply:
This should work like Java Strings. If the input is "abcd", the length would be 4. If the input is "\"abcd\"", the length would be 6.